frontend/src/lib/components/GroupFilter.svelte

```diff
@@ -0,0 +1,72 @@
+<script>
+    import { onMount } from 'svelte';
+    import { selectedGroupUuid, groupsMap } from '$lib/stores';
+    import { GetGroups } from '$wails/go/main/App';
+    import { getGroupColor } from '$lib/util';
+
+    export let items = [];
+    
+    onMount(async () => {
+        const backendGroups = await GetGroups();
+        
+        const mapping = new Map();
+        mapping.set(null, { uuid: null, name: "All", color: null });  
+        backendGroups?.forEach(group => {
+            mapping.set(group.uuid, {
+                uuid: group.uuid,
+                name: group.name,
+                color: getGroupColor(group.uuid)
+            });
+        });
+        
+        groupsMap.set(mapping);
+    });
```

Comment:
Prevent all entries from being filtered out when opening a file while a group filter is already set from a different file.

```suggestion
        // Reset the currently selected group when loading a file without a matching group
        if($selectedGroupUuid && !$groupsMap.has($selectedGroupUuid)) $selectedGroupUuid = null;
    });
```